app/features/worksheet_generator/tools.py

```diff
@@ -0,0 +1,481 @@
+from typing import List, Tuple, Dict, Any
+import os
+
+from langchain_core.pydantic_v1 import BaseModel, Field
```

Comment:
Refactor this to `from pydantic` as it is not compatible anymore with LangChain v. 0.3